### PR TITLE
Changed function Profile_User::getUserEntitiesForRight to return info…

### LINKED ISSUE #1720: [9.1.2] Function Profile_User::getUserEntitiesForRight doesn't work as expected
Hello,

This function can't retrieve what is expected, i.e.: the array of entities for which the given user has specific rights.

The $rights parameter is used to retrieve the field 'name' of the table, where it should be used to check the field 'rights'.

And a $name parameter is missing in function prototype.

Regards,
Tomolimo

### PR DESCRIPTION
**Pull request by tomolimo**
… as expected

Fixes #1720

### CODE REVIEW

**Comment by trasher** [comment]
This would introduce huge BC breaks; should be avoided in bugfixes.

**Comment by tomolimo** [comment]
This is really a bug fix, as it doesn't work at all today!!!

**Comment by tomolimo** [comment]
This function is not usable at all today, as it returns an empty array for any calls :(

**Comment by tomolimo** [comment]
And this function is only used in the generate ical feature of the planning.php file
could it be linked #1448?

**Comment by trasher** [comment]
> And this function is only used in the generate ical feature of the planning.php file
Since the method is public, it can be used in any plugin.

> could it be linked to #1448, #1422?
Absolutely no idea :D

**Comment by tomolimo** [comment]
> Since the method is public, it can be used in any plugin.

Yes, but not usable as it returns an empty array for each call

**Comment by orthagh** [comment]
For me, the change is effectively a breaking one but seems to be used only in core planning.
I have a huge list of plugin in my current glpi folder, none are using this method.

So the impact seems minor.

But as usual, any change in prototype function should not be in bugfixes branches.
The current function itself is correctly implemented, it's the call from planning which are bad.

So the good way to fix this issue:
- for 9.1/bugfixes: correct the planning call without alter the method prototype
- push a new pr for master to apply your changes

**Comment by tomolimo** [comment]
I've in my plans to do what @orthagh proposed :)

File: inc/profile_user.class.php [hunk 1]
```diff
@@ -654,12 +654,14 @@ static function getUserEntities($user_ID, $is_recursive=true, $default_first=fal
...
-    * @param $is_recursive              check also using recursive rights (true by default)
+    * @param $name            string    is the name of the rights to check (CommonDBTM::$rightname)
+    * @param $rights          integer   rights to check (may be a OR combinaison of several rights)
+    *                                   (exp: CommonDBTM::READ | CommonDBTM::UPDATE ...)
+    * @param $is_recursive    boolean   check also using recursive rights (true by default)
     *
     * @return array of entities ID
    **/
-   static function getUserEntitiesForRight($user_ID, $right, $is_recursive=true) {
+   static function getUserEntitiesForRight($user_ID, $name, $rights, $is_recursive=true) {
```
Comment (orthagh) [inline inc/profile_user.class.php]: maybe ``$rightname`` instead ``name`` (i know your not the author :) )

**Pull request closed without merging**